common/service/sensor/pldm_sensor.c

```diff
@@ -0,0 +1,353 @@
+						sensor_num, thread_id);
+				}
+			}
+			return;
+		}
+	}
+
+	if (pldm_sensor_cfg->post_sensor_read_hook) {
+		if (!pldm_sensor_cfg->post_sensor_read_hook(
+			    pldm_sensor_cfg, pldm_sensor_cfg->post_sensor_read_args, &reading)) {
+			pldm_sensor_cfg->cache_status = PLDM_SENSOR_FAILED;
+			*update_time = (k_uptime_get_32() / 1000);
+			LOG_DBG("Failed to pose read sensor_num 0x%x of thread %d", sensor_num,
+				thread_id);
+			return;
+		}
+	}
+
+	pldm_sensor_cfg->cache_status = PLDM_SENSOR_ENABLED;
+	*update_time = (k_uptime_get_32() / 1000);
+	pldm_sensor_cfg->cache = reading;
+}
+
+void pldm_sensor_polling_handler(void *arug0, void *arug1, void *arug2)
+{
+	ARG_UNUSED(arug1);
+	ARG_UNUSED(arug2);
+
+	int ret = 0, sensor_num = 0;
+	int thread_id = (int)arug0;
+	int pldm_sensor_count = 0;
+
+	pldm_sensor_count = plat_pldm_sensor_get_sensor_count(thread_id);
+	if (pldm_sensor_count <= 0) {
+		LOG_ERR("Failed to get PLDM sensor count(%d) of thread%d", pldm_sensor_count,
+			thread_id);
+		return;
+	}
+
+	pldm_sensor_list[thread_id] = plat_pldm_sensor_load(thread_id);
+	if (pldm_sensor_list[thread_id] == NULL) {
+		LOG_ERR("Failed to load PLDM sensor list of thread%d, ret%d", thread_id, ret);
+		return;
+	}
+
+	for (sensor_num = 0; sensor_num < pldm_sensor_count; sensor_num++) {
+		if (pldm_sensor_list[thread_id][sensor_num].pldm_sensor_cfg.pre_sensor_read_hook) {
+			if (!pldm_sensor_list[thread_id][sensor_num]
+				     .pldm_sensor_cfg.pre_sensor_read_hook(
+					     &pldm_sensor_list[thread_id][sensor_num]
+						      .pldm_sensor_cfg,
+					     pldm_sensor_list[thread_id][sensor_num]
+						     .pldm_sensor_cfg.pre_sensor_read_args)) {
+				pldm_sensor_list[thread_id][sensor_num]
+					.pldm_sensor_cfg.cache_status = PLDM_SENSOR_FAILED;
+				continue;
+			}
+		}
+
+		ret = sensor_drive_tbl[pldm_sensor_list[thread_id][sensor_num].pldm_sensor_cfg.type]
+			      .init(&pldm_sensor_list[thread_id][sensor_num].pldm_sensor_cfg);
+		if (ret < 0) {
+			pldm_sensor_list[thread_id][sensor_num].pldm_sensor_cfg.cache_status =
+				PLDM_SENSOR_FAILED;
+			LOG_ERR("Failed to init sensor 0x%x",
+				pldm_sensor_list[thread_id][sensor_num]
+					.pdr_numeric_sensor.sensor_id);
+		} else {
+			pldm_sensor_list[thread_id][sensor_num].pdr_numeric_sensor.sensor_init =
+				PDR_SENSOR_ENABLE;
+		}
+
+		if (pldm_sensor_list[thread_id][sensor_num].pldm_sensor_cfg.post_sensor_read_hook) {
+			if (!pldm_sensor_list[thread_id][sensor_num]
+				     .pldm_sensor_cfg.post_sensor_read_hook(
+					     &pldm_sensor_list[thread_id][sensor_num]
+						      .pldm_sensor_cfg,
+					     pldm_sensor_list[thread_id][sensor_num]
+						     .pldm_sensor_cfg.post_sensor_read_args,
+					     NULL)) {
+				pldm_sensor_list[thread_id][sensor_num]
+					.pldm_sensor_cfg.cache_status = PLDM_SENSOR_FAILED;
+				continue;
+			}
+		}
+	}
+
+	while (1) {
+		for (sensor_num = 0; sensor_num < pldm_sensor_count; sensor_num++) {
```

Comment:
Same thing for this for-loop. Move the body into its own function.
You could pass it &pldm_sensor_list[thread_id][sensor_num].

Reply:
Done.